swap/src/bin/swap.rs

```diff
@@ -222,8 +170,135 @@ async fn main() -> Result<()> {
             // Print the table to stdout
             table.printstd();
         }
-        Options::Resume { .. } => todo!("implement this"),
+        Command::Resume {
+            swap_id,
+            bitcoind_url,
+            bitcoin_wallet_name,
+            monero_wallet_rpc_url,
+            listen_addr,
+        } => {
+            let db_swap = db.get_state(swap_id)?;
+
+            if let Ok(alice_state) = AliceState::try_from(db_swap.clone()) {
+                let (bitcoin_wallet, monero_wallet) = setup_wallets(
+                    bitcoind_url,
+                    bitcoin_wallet_name.as_str(),
+                    monero_wallet_rpc_url,
+                    config,
+                )
+                .await?;
+                alice_swap(
+                    swap_id,
+                    alice_state,
+                    listen_addr,
+                    bitcoin_wallet,
+                    monero_wallet,
+                    config,
+                    db,
+                )
+                .await?;
+            } else if let Ok(bob_state) = BobState::try_from(db_swap) {
+                let (bitcoin_wallet, monero_wallet) = setup_wallets(
+                    bitcoind_url,
+                    bitcoin_wallet_name.as_str(),
+                    monero_wallet_rpc_url,
+                    config,
+                )
+                .await?;
+                bob_swap(swap_id, bob_state, bitcoin_wallet, monero_wallet, db).await?;
+            } else {
+                anyhow::bail!("Unable to construct swap state for swap with id {}")
+            }
+        }
     }
 
     Ok(())
 }
+
+async fn setup_wallets(
+    bitcoind_url: url::Url,
+    bitcoin_wallet_name: &str,
+    monero_wallet_rpc_url: url::Url,
+    config: Config,
+) -> Result<(Arc<bitcoin::Wallet>, Arc<monero::Wallet>)> {
+    let bitcoin_wallet =
+        bitcoin::Wallet::new(bitcoin_wallet_name, bitcoind_url, config.bitcoin_network)
+            .await
+            .expect("failed to create bitcoin wallet");
```

Comment:
```suggestion
            ?;
```

You are returning a result.